### ISSUE TITLE
Basic configuration type and schema validation.

### ISSUE THREAD

**Issue by toszo** [epiphany-platform/epiphany#195]
**Is your feature request related to a problem? Please describe.**
Currently there are almost not schema validations inside Epicli using jsonschema version 7.

**Describe the solution you'd like**
Add basic type and schema validation for the following folders containing jsonschema  validation documents:

- /workspaces/epiphany/core/src/epicli/data/azure/validation/infrastructure
- /workspaces/epiphany/core/src/epicli/data/aws/validation/infrastructure
- /workspaces/epiphany/core/src/epicli/data/azure/validation/infrastructure

**Describe alternatives you've considered**
None

**Additional context**
The following websites can be used to setup the basic structure and validation documents from the default documents.
1: https://www.json2yaml.com/convert-yaml-to-json - To convert YAML default to JSON
2: https://www.liquid-technologies.com/online-json-to-schema-converter - To convert the JSON defaults to JSON schema
3: https://www.json2yaml.com/ To convert JSON schema to YAML schema.

---

**DoD checklist**

* [x] Changelog updated
* [ ] COMPONENTS.md updated / doesn't need to be updated
* [x] Feature has automated tests
* [x] Automated tests passed (QA pipelines)
  * [x] apply
  * [ ] upgrade
* [ ] Idempotency tested
* [ ] Documentation added / updated / doesn't need to be updated
* [x] All conversations in PR resolved
* [x] Solution meets requirements and is done according to design doc
* [ ] Usage compliant with license
* [x] Backport tasks created / doesn't need to be backported

**Comment by sk4zuzu** [comment]
Hi there! :+1::innocent:

Here's my research about validation, a proof-of-concept very simple functional-style DSL https://github.com/sk4zuzu/RESEARCH/blob/master/jsonschema%2Bjmespath/main.py#L431-L448 that could be added to our schema documents and provide declarative dynamic validation of Epiphany's manfest (it's a working prototype).

You can see here that on top of the normal jsonchema validation we would add some more dynamic checking using an functonal (expression-only) DSL. :heart_eyes: 

A working example (it validates the schema first and then runs "dynamic" checks):
```yaml
type: object
required:
  - "cluster"
  - "infra/vm"
properties:
  "cluster":
    type: object
    patternProperties:
      "^[a-z0-9]+$":
        required:
          - spec
---
all:
  - eq:
      - jmespath: >-
          "infra/vm".{{ cluster.* | [0].spec.app1.vm }}.spec.os
      - jmespath: >-
          "infra/vm".{{ cluster.* | [0].spec.app2.vm }}.spec.os
```

Why additional DSL? That's because jsonschema does not support such functionality and it seems it's unlikely it will. Also I was unable to find anything like this... :'C

So idea would be (thank you @seriva!) to make it a small separate stalically-linked binary/command in golang or rust (using musl toolchain), make it public and then reuse it in Epiphany or wherever we would need. :)

WDYT? :thinking:

**Comment by seriva** [comment]
> Hi there! 👍😇
> 
> Here's my research about validation, a proof-of-concept very simple functional-style DSL https://github.com/sk4zuzu/RESEARCH/blob/master/jsonschema%2Bjmespath/main.py#L431-L448 that could be added to our schema documents and provide declarative dynamic validation of Epiphany's manfest (it's a working prototype).
> 
> You can see here that on top of the normal jsonchema validation we would add some more dynamic checking using a functonal (expression-only) DSL. 😍
> 
> A working example (it validates the schema first and then runs "dynamic" checks):
> 
> ```yaml
> type: object
> required:
>   - "cluster"
>   - "infra/vm"
> properties:
>   "cluster":
>     type: object
>     patternProperties:
>       "^[a-z0-9]+$":
>         required:
>           - spec
> ---
> all:
>   - eq:
>       - jmespath: >-
>           "infra/vm".{{ cluster.* | [0].spec.app1.vm }}.spec.os
>       - jmespath: >-
>           "infra/vm".{{ cluster.* | [0].spec.app2.vm }}.spec.os
> ```
> 
> Why additional DSL? That's because jsonschema does not support such functionality and it seems it's unlikely it will. Also I was unable to find anything like this... :'C
> 
> So idea would be (thank you @seriva!) to make it a small separate stalically-linked binary/command in golang or rust (using musl toolchain), make it public and then reuse it in Epiphany or wherever we would need. :)
> 
> WDYT? 🤔

Not relevant for this issue, but ill link it to the one that might deal with this eventually:)

**Comment by przemyslavic** [comment]
Validation works fine with the fixes made in #2474.
